common/zone_store.cpp

```diff
@@ -152,11 +154,35 @@ ZoneRepository::Zone ZoneStore::GetZone(uint32 zone_id, int version)
  */
 ZoneRepository::Zone ZoneStore::GetZone(const char *in_zone_name)
 {
-	for (auto &z: zones) {
+	for (auto &z: m_zones) {
 		if (z.short_name == in_zone_name) {
 			return z;
 		}
 	}
 
-	return ZoneRepository::Zone();
+	return ZoneRepository::NewEntity();
+}
+
+const std::vector<ZoneRepository::Zone> &ZoneStore::GetZones() const
+{
+	return m_zones;
+}
+
+// gets zone data by using explicit version and falling back to version 0 if not found
+ZoneRepository::Zone ZoneStore::GetZoneWithFallback(uint32 zone_id, int version)
```

Comment:
This function is returning a copy of Zone, which doesn't seem very efficient.

Reply:
Agreed. Although - if it was easily able to be returned by reference or pointer we would here, but we are not always guaranteed we'll find an entity so we return an entirely new default entity so related object can determine it doesn't exist (z.id == 0) etc. 

Even with returning a memory copy this will be infinitely more efficient than all the silly database calls

Reply:
If you can return const ref, you can initialize a function level static and return that in the case a value doesn't exist.

Reply:
another option is to return a pointer, and return nullptr if it doesn't exist.

Reply:
Could explore different models. I’m out for the night for now. Need to get ready for the week